velox/vector/FlatVector.h

```diff
@@ -350,7 +351,7 @@ class FlatVector final : public SimpleVector<T> {
 
   // If T is velox::StringView, the referenced is held by
   // one of these.
-  std::vector<BufferPtr> stringBuffers_;
+  folly::F14VectorSet<BufferPtr> stringBuffers_;
```

Comment:
What is the advantage of making it a set?

Reply:
Acquire string buffer operation is a linear search with vector type which might be potential performance issue when gather copy from a large number of source buffers.

Reply:
Can we convert the `vector` to set only in `acquireSharedStringBuffers`?  That way you can achieve the same time complexity while keeping the interface simple (and avoid the overhead of using a set).

Reply:
Yeah, but Set doesn't provide the first and last element access.

Reply:
The current hack to use `begin()` and `rbegin()` is also not very safe.  I would suggest keep the `vector` interface and create a temporary `F14VectorSet` (or in-place sort) inside `acquireSharedStringBuffers()` to avoid the N^2 lookup.

Reply:
Per offline discussion, switch to use F14FastSet internally to keep track of the referenced raw buffer pointer for this. Thanks for the suggestion.